Merge 4.6 into checkpoint feature branch 2020-05-05

lankydan opened the pull request.

przemolb committed:
Add debug dockerfiles with additional tools useful for investigating network issues.

przemolb committed:
Merge pull request #5610 from corda/pb-add-docker-with-debug-tools

Add debug dockerfiles with additional tools useful for investigating …

josephzunigadaly committed:
ENT-4857: Fix race condition in trackTransaction (#6096)

- Fix issue
- Emit warning if we are inside a DB transaction
- Include a path that does not emit warning
- Add unit tests

Commit:
ENT-4659: Remove CryptoServiceFactory from OS

kyriathar-r3 committed:
CORDA-3677 FlowExternalOperation serialising reference to FlowLogic (#6094)

* Stop capturing 'FlowLogic' references in flowAsyncOperation;

Creating concrete classes removes the implicit reference to FlowLogic (as this) being included in the anonymous object

* Modify test code so that lambdas no longer get implicit references to their enclosing 'FlowLogic'

rick-r3 committed:
Merge pull request #6103 from corda/dan/backport-corda-3677

Backport CORDA-3677

Commit:
Merge pull request #6100 from corda/denis/ENT-4659-remove-crypto-service-factory

ENT-4659: Remove CryptoServiceFactory from OS

josephzunigadaly committed:
Merge remote-tracking branch 'origin/release/os/4.3' into jzd/merge-os4.3-into-os4.4-2020-03-26

rick-r3 committed:
Merge pull request #6104 from corda/jzd/merge-os4.3-into-os4.4-2020-03-26

[NOTICK] - Merge OS 4.3 into OS 4.4 (2020-03-26)

lankydan committed:
Merge branch 'release/os/4.4' into dan/os-4.4-to-4.5-merge-2020-03-26

dimosr committed:
[NOTICK] - Enable check in detekt for unused imports (#6106)

* [NOTICK] - Enable check in detekt for unused imports

* Put back accidental removal of used import

* Some more accidental removals

chrisr3 committed:
CORDA-3680: Add CorDapp custom serialisers to Driver's in-process nodes. (#6102)

* Run serialisation tests with both in-process and out-of-process nodes.

* Add custom serialisers and whitelists to Driver's AMQPServerSerializationScheme.

lankydan committed:
Merge pull request #6108 from corda/dan/os-4.4-to-4.5-merge-2020-03-26

OS 4.4 to 4.5 merge 2020-03-26

chrisr3 committed:
CORDA-3680: Add CorDapp custom serialisers to Driver's in-process nodes. (#6098)

* Run serialisation tests with both in-process and out-of-process nodes.

* Add custom serialisers and whitelists to Driver's AMQPServerSerializationScheme.

blsemo committed:
ENT-5134 Discoverable Hibernate Session Factory Factory (#6091)

* Introduce CordaSessionFactoryFactory interface and the H2 implememntation
* Load SessionFactoryFactory via service loader
* Add Postgres SessionFactoryFactory
* Add extraConfiguration function for SessionFactoryFactory implementations to expose special config values.

fowlerrr committed:
CORDA-3675: Update netty for security (#6109)

* CORDA-3675: Update netty for security

* Try to improve thread counting in tests

chrisr3 committed:
ENT-4967: Allow quasar_classifier to be either null or an empty string. (#6119)

dgounaris committed:
TM-213: Add sonarqube configuration in gradle and jenkinsfile (#6060)

* Add sonarqube for regression builds

* Remove duplicated script tag

* Add detekt support to sonarqube

Co-authored-by: Dimitris Gounaris <dimitrisgounaris@20LDN-MAC257.local>

chrisr3 committed:
CORDA-3643: Initialise MockNetwork's serialisation environment with all custom cordapps. (#6115)

fowlerrr committed:
Revert "CORDA-3675: Update netty for security (#6109)" (#6120)

This reverts commit 82a3f69f

StefanIliev545 committed:
Notary logs warning on startup when it's not in the whitelist (#5857)

chrisr3 committed:
ENT-5039 Improved help text for commands (#6006) (#6122)

* Improved help text for commands
* Address feedback

Co-authored-by: jakubbielawa <jakub.bielawa@gmail.com>

chrisr3 committed:
CORDA-3688: Apply @Named annotation to CRaSH commands to fix usage messages. (#6124)

chrisr3 committed:
CORDA-3688: Apply @Named annotation to CRaSH commands to fix usage messages. (#6126)

dimosr committed:
[ENT-4754] - Open access to members of FinalityFlow (#6113)

* [ENT-4754] - Open access to members of FinalityFlow

* Adjust slightly to provide access without changing the public api

dgounaris committed:
Make sonarqube step resilient to errors (#6128)

Co-authored-by: Dimitris Gounaris <dimitrisgounaris@20LDN-MAC257.local>

adelel1 committed:
CORDA-3690: Changed algorithm name used in signature scheme object from ECDSA to EC… (#6123)

* CORDA-3690: Changed algorithm name used in signature from ECDSA to EC. JDK11 checks with in key generation.

* CORDA-3690: Remove the SHA512WITHSPHINCS256 signature scheme from the generate key pair and sign test.

* CORDA-3690: Algorithm in SignatureScheme has changed to EC from ECDSA so change test to match.

Co-authored-by: Adel El-Beik <adelel-beik@19LDN-MAC108.local>

Commit:
ENT-5089: Forcibly free associated with transport resources if it is already closed (#6130)

woggioni committed:
fixed bug in error reporting (#6127)

the current code doesn't forward the exception to the logging system, this means that any cause inside the exception is lost as with all of the stacktraces (both the one of the thrown exception and the one belonging to its cause).
The correct way to log an exception is to pass both the message and the exception to the logging system.

chrisr3 committed:
CORDA-3689: Fix scanning AppClassLoader on Java 11. (#6121)

* Upgrade ClassGraph to 4.6.68 to fix scanning AppClassLoader on JDK11.
* Revert CORDA-1575 to prevent CordaRPCClient logs from being clobbered.

chrisr3 committed:
CORDA-3698: Require no classifier for Open Core and DJVM-related modules. (#6132)

* ENT-4967: Require no classifier for corda-node-djvm, corda-deserializers-djvm.

* Also remove classifiers from core, serialization and finance-contracts.

* Compile corda-serialization-djvm for Java 8 and remove its classifier.

dgounaris committed:
Add github actions to sync open and closed issues in jira (#6138)

Co-authored-by: Dimitris Gounaris <dimitrisgounaris@20LDN-MAC257.local>

dgounaris committed:
Add github action for syncing assignees between github-jira (#6143)

Co-authored-by: Dimitris Gounaris <dimitrisgounaris@20LDN-MAC257.local>

wzur-r3 committed:
Update README.md (#6144)

wzur-r3 committed:
Changed frequency of Github Actions for sync between GH and Jira (#6148)

* run every hour instead every 15 minutes
* assign & close executed at different times

chrisr3 committed:
CORDA-3562: Update SandboxToStringSerializer not to need extra sandbox task. (#6149)

fowlerrr committed:
Corda 3675 update netty (#6146)

* CORDA-3675: Update netty for security

* Try to improve thread counting in tests

Schife committed:
INFRA-284 Testing out new aks template (#6150)

* INFRA-284 Testing out new aks template
* INFRA-284 switching back to k8s on all files

adelel1 committed:
CORDA-3696: JDK 11 Testing branch (#6131)

* CORDA-3696: Temporary update to enable JDK11 build and test. Will eventually be switchable.

* CORDA-3696: Filter out the Nashorn warning.

* CORDA-3696: Add JDK11 classifier.

* CORDA-3696: Updated match string to cope with JDK11.

* CORDA-3696: Filtering out SPHINCS256_SHA256 where failing due to JDK11.

* CORDA-3696: Now remove SPHINCS256_SHA256 only if JDK11.

* CORDA-3696: Fix test failure - switch to regex matching.

* CORDA-3696: Hide the illegal access warnings.

* CORDA-3696: Check for Java11 when disabling Java11 warnings.

* CORDA-3696: Fix unneccessary non null check.

* CORDA-3696: Reverting build env to JDK8

* CORDA-3696: Revert hiding of illegal access warnings via Unsafe class.

* CORDA-3696: Remove internal access warnings and new JDK11 version checker.

* CORDA-3696: Updated build file for OS

* CORDA-3696: Removed typo

* CORDA-3696: Fixed space typo.

* CORDA-3696: Open modules to remove the illegal access warnings.

Co-authored-by: Adel El-Beik <adelel-beik@19LDN-MAC108.local>

relyafi committed:
[EG-503] Spent state audit tool (#6107)

* [EG-503] Spent state audit tool

Fixes

* Refinements to notary query interfaces. Feature complete.

* EG-503: Introduce optional `notaryService` in `ServiceHubCoreInternal`

* Remove redundant logic following change to use extensions API

Co-authored-by: Viktor Kolomeyko <viktor.kolomeyko@r3.com>

Schife committed:
INFRA-284 switching label (#6153)

Schife committed:
INFRA-284 switching from local k8s label (#6157)

Schife committed:
INFRA-284 switching from local k8s label (#6156)

relyafi committed:
[EG-503] Test cases for exposed notaryService (#6161)

chrisr3 committed:
CORDA-3716: Fix SandboxEnumSerializer to handle enums that override toString(). (#6159)

* CORDA-3716: Fix SandboxEnumSerializer to handle enums that override toString().

* Remove more uses of Enum.toString() from the Corda serializer.

* Add test coverage for this case to standard enum serializer.

* Increase maxWaitTimeout in IRSDemoTest to 150 seconds.

dimosr committed:
[ENT-5210] - Whitelist SNAPPY encoding (#6163)

* [ENT-5210] - Whitelist SNAPPY encoding

* Remove unused imports

chrisr3 committed:
CORDA-3716: Fix Enum serializers to handle enums that override toString()

* CORDA-3716: Fix SandboxEnumSerializer to handle enums that override toString().

* Remove more uses of Enum.toString() from the Corda serializer.

* Add test coverage for this case to standard enum serializer.

* Increase maxWaitTimeout in IRSDemoTest to 150 seconds.

blsemo committed:
ENT-4494 revert postgre to previous open source behaviour (#6166)

* ENT-4494 revert to previous open source behaviour re postgres while discussions about DB harmonization is ongoing

* Remove unused imports

nikinagy committed:
CORDA-3569 - Add RestrictedConnection and more blocked methods to RestrictedEntityManager (#6129)

* adding blocked functions ro RestrictedEntityManager and creating RestrictedConnection class

* adding flow tests and fixing issues regarding the review

* adding quasar util to gradle

* updating flow tests

* adding space before } at .isThrownBy()

* adding spaces

dimosr committed:
[NOTICK] - Bump platform version to 7

Commit:
ENT-4912: Enable CRL checking with embedded Artemis (#6154)

nikinagy committed:
making sure hibernate uses UTC time zone (#6168)

rick-r3 committed:
CORDA-3701 Fix bugs in some iterator checkpoint serializers (#6135)

* CORDA-3701 Fix bugs in some iterator checkpoint serializers

* Added some more tests and tidied up implementation some more.

* Fix imports to be detekt compliant

* Add timeouts to tests

adelel1 committed:
CORDA-3730: New version enabled features can now be switched on with constants. (#6175)

Commit:
CORDA-3726: Remove memory leaks when many InProcess nodes started (#6169)

* CORDA-3762: Integration test exposing the problem reported

* CORDA-3726: Additional logging

* CORDA-3726: Prevent thread leaks

* CORDA-3726: New `journalBufferTimeout` parameter

* CORDA-3726: Override `journalBufferTimeout` parameter

* CORDA-3726: Making Detekt happier

* CORDA-3276: Account for extra thread user in MockNetwork

For real node this does not matter as `shutdown` can safely be called multiple times, which is not true for server thread provided by MockNetwork

* CORDA-3276: Do not make SMM shutdown "executor" as it belongs to AbstractNode

* CORDA-3276: Address input from @rick-r3

* CORDA-3276: Fix test after rebase

josephzunigadaly committed:
CORDA-3316: Add ThrowableEvolutionTests unit test (#6167)

* CORDA-3316: Add ThrowableEvolutionTests unit test

* Add timeouts to tests

* Remove extra space before class name

ivanterziev-r3 committed:
EG-1404 - Retired the rst doc source structure under /docs and update… (#6147)

* Retired the rst doc source structure under /docs and updated the /docs/README.md
*Rollback of /example-code and /whitepaper dirs back under /docs dir until new code example process is in place

Commit:
ENT-5219: Synchronize BCCryptoService between OS and ENT (#6178)

josephzunigadaly committed:
ENT-5141: Fix ConcurrentModificationException in FetchDataFlow (#6176)

* ENT-5141: Fix ConcurrentModificationException in FetchDataFlow

* Make detekt happy

* Make CheckpointSerializationEnvironmentRule inheritable

nikinagy committed:
Merge branch 'release/os/4.3' into nnagy-os-4.3-os-4.4-20200423

mvdbos committed:
Load CordaServices before NotaryService (#6173)

On node start, load CordaServices before starting the NotaryService,
so that the NotaryService can check that the services it requires are
available when starting.

Resolves #6172.

Commit:
ENT-5222: Prevent from blocking further connections after transport failure on TLS handshake

mnesbit committed:
Merge pull request #6181 from corda/nnagy-os-4.3-os-4.4-20200423

Nnagy os 4.3 os 4.4 20200423

nikinagy committed:
Merge branch 'release/os/4.4' of https://github.com/corda/corda into nnagy-os4.4-os4.5-20200426-2

 Conflicts:
	node-api/src/main/kotlin/net/corda/nodeapi/internal/persistence/HibernateConfiguration.kt
	node-api/src/main/kotlin/net/corda/nodeapi/internal/rpc/client/AMQPClientSerializationScheme.kt

nikinagy committed:
Add Hibernate UTC Fix

josephzunigadaly committed:
Move files to their new home

Commit:
Merge pull request #6183 from corda/denis/ENT-5222-tls-handshake-error

ENT-5222: Prevent from blocking connections after transport failure on TLS handshake

xavierlepretre committed:
Make TransactionBuilder's input references deep copied on copy.

rick-r3 committed:
Merge pull request #6185 from corda/nnagy-os4.4-os4.5-20200426-2

os4.4->os4.5 2020-04-26

nikinagy committed:
CORDA-3659 - adding kdoc for RestrictedEntityManager and RestrictedConnection (#6179)

* adding kdoc for RestrictedEntityManager and RestrictedConnection

* adding kdoc for RestrictedEntityManager and RestrictedConnection

lankydan committed:
CORDA-3722 withEntityManager can rollback its session (#6187)

* CORDA-3722 withEntityManager can rollback its session

## Summary

Improve the handling of database transactions when using
`withEntityManager` inside a flow.

Extra changes have been included to improve the safety and
correctness of Corda around handling database transactions.

This focuses on allowing flows to catch errors that occur inside an
entity manager and handle them accordingly.

Errors can be caught in two places:

- Inside `withEntityManager`
- Outside `withEntityManager`

Further changes have been included to ensure that transactions are
rolled back correctly.

## Catching errors inside `withEntityManager`

Errors caught inside `withEntityManager` require the flow to manually
`flush` the current session (the entity manager's individual session).
By manually flushing the session, a `try-catch` block can be placed
around the `flush` call, allowing possible exceptions to be caught.

Once an error is thrown from a call to `flush`, it is no longer possible
to use the same entity manager to trigger any database operations. The
only possible option is to rollback the changes from that session.
The flow can continue executing updates within the same session but they
will never be committed. What happens in this situation should be handled
by the flow. Explicitly restricting the scenario requires a lot of effort
and code. Instead, we should rely on the developer to control complex
workflows.

To continue updating the database after an error like this occurs, a new
`withEntityManager` block should be used (after catching the previous
error).

## Catching errors outside `withEntityManager`

Exceptions can be caught around `withEntityManager` blocks. This allows
errors to be handled in the same way as stated above, except the need to
manually `flush` the session is removed. `withEntityManager` will
automatically `flush` a session if it has not been marked for rollback
due to an earlier error.

A `try-catch` can then be placed around the whole of the
`withEntityManager` block, allowing the error to be caught while not
committing any changes to the underlying database transaction.

## Savepoints / Transactionality

To make `withEntityManager` blocks work like mini database transactions,
save points have been utilised. A new savepoint is created when opening
a `withEntityManager` block (along with a new session). It is then used
as a reference point to rollback to if the session errors and needs to
roll back. The savepoint is then released (independently from
completing successfully or failing).

Using save points means, that either all the statements inside the
entity manager are executed, or none of them are.

## Some implementation details

- A new session is created every time an entity manager is requested,
but this does not replace the flow's main underlying database session.
- `CordaPersistence.transaction` can now determine whether it needs
to execute its extra error handling code. This is needed to allow errors
escape `withEntityManager` blocks while allowing some of our exception
handling around subscribers (in `NodeVaultService`) to continue to work.

nikinagy committed:
switch on the publishing of sources

JamesHR3 committed:
[EG-438] Error Reporting Framework (#6125)

* [EG-438] First commit of error code interface

* [EG-438] Implement error reporter and a few error codes

* [EG-438] Add unit tests and default properties files

* [EG-438] Add the error table builder

* [EG-438] Update initial properties files

* [EG-438] Add some Irish tests and the build.gradle

* [EG-438] Fall back for aliases and use different resource strategy

* [EG-438] Define the URL using a project-specific context

* [EG-438] Tidy up initialization code

* [EG-438] Add testing to generator and tidy up

* [EG-438] Remove direct dependency on core and add own logging config

* [EG-438] Fix compiler warnings and tidy up logging

* [EG-438] Fix detekt warnings

* [EG-438] Improve error messages

* [EG-438] Address first set of review comments

* [EG-438] Use enums and a builder for the reporter

* [EG-438] Add kdocs for error resource static methods

* [EG-438] Handle enums defined with underscores

* [EG-438] Slight refactoring of startup code

* [EG-438] Port changes to error reporting code from future branch

* [EG-438] Also port test changes

* [EG-438] Suppress a deliberately unused parameter

chrisr3 committed:
CORDA-3738: Upgrade to DJVM 1.1-RC01. (#6182)

* Update for changes to API for DJVM 1.1-RC01.

* Tidy up generics for DJVM serialization.

lankydan committed:
CORDA-3291 `isKilled` flag and session errors for killed flows (#6170)

* CORDA-3291 `isKilled` flag and session errors for killed flows

## Summary

Two major improvements have been worked on:

- A new flag named `isKilled` has been added to `FlowLogic` to allow
developers to break out of loops without suspension points.
- Killed flows now send session errors to their counter parties allowing
their flows to also terminate without further coordination.

Achieving these changes required a __fundamental__ change to how flows are
killed as well as how they sleep.

## `isKilled` flag

The addition of `FlowLogic.isKilled` allows flows to check if the
current flow has been killed. They can then throw an exception to lead
to the flow's termination (following the standard error pathway). They
can also perform some extra logic or not throw an exception if they
really wanted to.

No matter what, once the flag is set, the flow will terminate. Due to
timing, a killed flow might successfully process its next suspension
event, but it will then process a killed transition and terminate.

## Send session errors when killing a flow

A flow will now send session errors to all of its counter parties. They
are transferred as `UnexpectedFlowEndException`s. This allows initiated
flows to handle these errors as they see fit, although they should
probably just terminate.

## How flows are killed

### Before

Originally we were relying on Quasar to interrupt a flow's fiber, we
could then handle the resulting `InterruptedException`. The problem with
this solution is that it only worked when a flow was already suspended
or when a flow moved into suspension. Flows stuck in loops did not work.

### After

We now *do not* use Quasar to interrupt a flow's fiber. Instead, we
switch `FlowStateMachine.isKilled` to true and schedule a new event.
Any event that is processed after switching this flag will now cause a
`KilledFlowTransition`. This transition follows similar logic to how
error propagation works. Note, the extra event allows a suspended flow
to be killed without waiting for the event that it was _really_ waiting
for.

This allows a lot of the tidy up code in `StateMachineManager.killFlow`
to be removed as tidy up is executed as part of removing a flow.
Deleting a flow's checkpoint and releasing related soft locks is still
handled manually in case of infinite loops but also triggered as part
of the actions executed in a transition.

This required flow sleeping to be changed as we no longer rely on
quasar.

## How flows now sleep

The reliance on Quasar to make a flow sleep has been removed.

Instead, when a flow sleeps we create a `ScheduledFuture` that is
delayed for the requested sleep duration. When the future executes it
schedules a `WakeUpFromSleep` event that wakes up the flow... Duh.

`FlowSleepScheduler` handles the future logic. It also uses the same
scheduled thread pool that timed flows uses.

A future field was added to `StateMachineState`. This removes the 
need for concurrency control around flow sleeps as the code path does
not need to touch any concurrent data structures.

To achieve this:

- `StateMachineState.future` added as a `var`
- When the `ScheduledFuture` is created to wake up the flow the passed
in `StateMachineState` has its `future` value changed
- When resumed `future` and `isWaitingForFuture` are set to `null` and
`false` respectively
- When cancelling a sleeping flow, the `future` is cancelled and nulled
out. `isWaitingForFuture` is not changed since the flow is ending anyway
so really the value of the field is not important.

chrisr3 committed:
CORDA-3685: Prevent ServiceHub.signInitialTransaction from throwing undeclared checked exceptions. (#6111)

JamesHR3 committed:
[EG-440] Add some error codes and the error resource generation tool (#6192)

* [EG-438] First commit of error code interface

* [EG-438] Implement error reporter and a few error codes

* [EG-438] Add unit tests and default properties files

* [EG-438] Add the error table builder

* [EG-438] Update initial properties files

* [EG-438] Add some Irish tests and the build.gradle

* [EG-438] Fall back for aliases and use different resource strategy

* [EG-438] Define the URL using a project-specific context

* [EG-438] Tidy up initialization code

* [EG-438] Add testing to generator and tidy up

* [EG-438] Remove direct dependency on core and add own logging config

* [EG-438] Fix compiler warnings and tidy up logging

* [EG-438] Fix detekt warnings

* [EG-438] Improve error messages

* [EG-438] Address first set of review comments

* [EG-438] Use enums and a builder for the reporter

* [EG-438] Address first set of review comments

* [EG-438] Use enums and a builder for the reporter

* [EG-438] Add kdocs for error resource static methods

* [EG-440] Add error code for duplicate CorDapp loading

* [EG-438] Handle enums defined with underscores

* [EG-440] Add errors for some CorDapp loading scenarios

* [EG-440] Finish adding errors for CorDapp loading

* [EG-440] Fix up errors in properties files

* [EG-440] Start change to error code definition

* [EG-440] Update error code definition and add resource generation tool

* [EG-440] Tidy up error resource generation tool frontend

* [EG-440] Small refactorings and add kdocs

* [EG-440] Generate all missing resources

* [EG-440] Some refactoring and start writing a test

* [EG-440] Update unit test for resource generator

* [EG-440] Renaming of various parts of the error tool

* [EG-440] Add testing for errors and fix an issue in resource generation

* [EG-440] Add a kdoc for context provider API

* [EG-440] Remove old code from repository

* [EG-440] Address some review comments

JamesHR3 committed:
Merge branch 'release/os/4.5' into jamesh/error-reporting-sync-29-04-20

# Conflicts:
#	node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt

JamesHR3 committed:
[MERGE] Fix detekt issues

JamesHR3 committed:
Merge pull request #6196 from corda/jamesh/error-reporting-sync-29-04-20

[MERGE] Update Error Reporting feature branch to latest release/os/4.5

mnesbit committed:
Merge pull request #6190 from corda/nikinagy/CORDA-3747-publish-sources-for-node-driver

CORDA-3747 - switch on the publishing of sources

adelel1 committed:
CORDA-3715: Check contract classes hav… (#6155)

* CORDA-3715: When loading cordapps now check that contract classes have class version between 49 and 52

* CORDA-3715: Now check class version when contract verification takes place.

* CORDA-3715: Making detekt happy with number of levels in func

* CORDA-3715: Make use of new ClassGraph release which provides class file major version number.

* CORDA-3715: Changed package name in test jars

* CORDA-3715: Use ClassGraph when loading attachments.

* CORDA-3715: Reverted file to 4.5 version

* CORDA-3715: Updating method to match non deterministic version.

* CORDA-3715: Added in default param.

* CORDA-3715: Adjusted min JDK version to 1.1

* CORDA-3715: Switching check to JDK 1.2

* CORDA-3715: Now version check SerializationWhitelist classes.

* CORDA-3715: Switched default to null for range.

Schife committed:
INFRA-311 new jdk11 jenkinsfile (#6194)

lankydan committed:
CORDA-3291 Support for new flow sleep in mock networks (#6200)

The new flow sleep made `NotaryWhitelistTests` flaky which highlighted
an issue with the change to how a flow sleeps. Messages were being
pumped while the flow was sleeping which lead to inconsistent behaviour.
Messages are no longer pumped while a flow sleeps.

lankydan committed:
CORDA-3291 Fix broken tests (#6197)

lankydan committed:
NOTICK Ignore JUnit time threads in `RPCStabilityTests` (#6201)

JamesHR3 committed:
[NOTICK] Fix a test error to allow tests to pass in Jenkins (#6202)

adelel1 committed:
CORDA-3191: When querying for attachment ids now only retrieve the attachment id columm not whole object. (#6199)

chrisr3 committed:
CORDA-3738: Upgrade to DJVM 1.1-RC02. (#6195)

* CORDA-3738: Upgrade to DJVM 1.1-RC02.

* Update comment for DJVM 1.1's new requirements.

Commit:
EG-433 - Error Reporting Framework (Merge)

Error Reporting Framework - Phase 1

lankydan committed:
Revert "NOTICK Ignore JUnit time threads in `RPCStabilityTests` (#6201)" (#6204)

This reverts commit ecf74053

josephzunigadaly committed:
ENT-5237: Remove DISTINCT because Oracle cannot apply DISTINCT to BLOB fields (#6203)

chrisr3 committed:
CORDA-3745: Modify DJVM serializers to support Enum Evolution. (#6189)

jmacmahonr3 committed:
EG-153 - Adding sshd by default to docker node.conf template (#6171)

* Adding value needed for the rapid deployment of Azure UAT enviroments
* Parameterizing Rpc user
* Removing duplicate config requirement
* Forcing users to provide sshPort for node config generation with docker image

xavierlepretre committed:
Reverted comment change about immutable accessors.

nargas-ritu committed:
Merge pull request #6186 from xavierlepretre/release/os/4.5

CORDA-3754 Make TransactionBuilder's input references deep copied on copy.

chrisr3 committed:
CORDA-3758: Ensure that providerMap is immutable, at the very least. (#6209)

chrisr3 committed:
CORDA-3750: Delete unwanted methods from the deterministic artifacts. (#6208)

nargas-ritu committed:
Bump OS release version 4.6

lankydan committed:
Merge branch 'release/os/4.6' into dan/4.6-into-checkpoint-feature-branch-2020-05-05

# Conflicts:
#	node/src/main/kotlin/net/corda/node/services/statemachine/Event.kt
#	node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt
#	node/src/main/kotlin/net/corda/node/services/statemachine/SingleThreadedStateMachineManager.kt
#	node/src/main/kotlin/net/corda/node/services/statemachine/transitions/TopLevelTransition.kt

lankydan committed:
NOTICK Fix `KilledFlowTransition` after merge

lankydan committed:
NOTICK Remove unused imports

williamvigorr3 commented:
Looks fine to me. I locally merged the branches and both rebases I need to do look pretty easy.

lankydan committed:
NOTICK Fix kill flow tests due to storing failed flows

Failed flows are stored after the checkpoint table rework. This meant
that some of the asserts in `FlowIsKilledTest` and `KillFlowTest` were
wrong.

lankydan committed:
NOTICK Change platform version assert in `FlowMetadataRecordingTest`
